cmd/mount2/node.go

```diff
@@ -404,3 +404,83 @@ func (n *Node) Rename(ctx context.Context, oldName string, newParent fusefs.Inod
 }
 
 var _ = (fusefs.NodeRenamer)((*Node)(nil))
+
+// Getxattr should read data for the given attribute into
+// `dest` and return the number of bytes. If `dest` is too
+// small, it should return ERANGE and the size of the attribute.
+// If not defined, Getxattr will return ENOATTR.
+func (n *Node) Getxattr(ctx context.Context, attr string, dest []byte) (nbytes uint32, errno syscall.Errno) {
+	defer log.Trace(n, "attr=%q, len(dest)=%d", attr, len(dest))("errno=%v", &errno)
+	data, err := n.node.Getxattr(attr)
+	if err != nil {
+		return 0, translateError(err)
+	}
+
+	var datalen int = len(data)
+	nbytes = uint32(datalen)
+	if int(nbytes) != datalen {
+		errno = syscall.E2BIG
+		return
+	}
+
+	if datalen > len(dest) {
+		errno = syscall.ERANGE
+		return
+	}
+
+	copy(dest, data)
```

Comment:
If we change the API of n.node.Getxattr to take a non-reallocateable slice, it moves this complexity into the vfs layer, but is more performant (we can skip this copy). Do you have a preference on simplicity vs. performance in the vfs layer?